### PR TITLE
Add error type and handle it in bot creation process

### PR DESCRIPTION
**Pull request by cojoj**
This is a **fix** to the #114 👌

### CODE REVIEW

File: XcodeServerSDK/API Routes/XcodeServer+Bot.swift [hunk 1]
```diff
@@ -50,8 +50,12 @@ extension XcodeServer {
                     return
                 }
                 
-                //we have platforms, find the one in the bot config and replace it
-                self.replacePlaceholderPlatformInBot(botOrder, platforms: platforms!)
+                do {
+                    //we have platforms, find the one in the bot config and replace it
+                    try self.replacePlaceholderPlatformInBot(botOrder, platforms: platforms!)
+                } catch {
+                    completion(response: .Error(error: error))
```
Comment (czechboy0) [inline XcodeServerSDK/API Routes/XcodeServer+Bot.swift:57]: You also need a return otherwise it would keep going and try to create the Bot anyway.

Comment (cojoj) [inline XcodeServerSDK/API Routes/XcodeServer+Bot.swift:57]: Ok, just pushed the `return` 😉

**Comment by buildasaur** [comment]
Result of Integration **1**
---
*Duration*: 1 minute and 56 seconds
*Result*: **-1 error, failing state: trigger-error**

**Comment by buildasaur** [comment]
Result of Integration **2**
---
*Duration*: 1 minute and 36 seconds
*Result*: **-1 error, failing state: trigger-error**

**Comment by buildasaur** [comment]
## Result of Integration **3**

_Duration_: 1 minute and 2 seconds
_Result_: **Perfect build!** All 73 tests passed. :+1:
_Test Coverage_: 56%.

**Comment by czechboy0** [comment]
Thanks, @cojoj!

**Pull request merged**